slug_test.go

```diff
@@ -240,6 +240,79 @@ func TestPackWithoutDereferencing(t *testing.T) {
 	}
 }
 
+func TestPackWithoutIgnoring(t *testing.T) {
+	slug := bytes.NewBuffer(nil)
+
+	// By default NewPacker() creates a Packer that does not use
+	// .terraformignore or dereference symlinks.
+	p, err := NewPacker()
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	meta, err := p.Pack("testdata/archive-dir", slug)
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	gzipR, err := gzip.NewReader(slug)
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	tarR := tar.NewReader(gzipR)
+	var (
+		fileList []string
+		slugSize int64
+	)
+
+	for {
+		hdr, err := tarR.Next()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			t.Fatalf("err: %v", err)
+		}
+
+		fileList = append(fileList, hdr.Name)
+		if hdr.Typeflag == tar.TypeReg || hdr.Typeflag == tar.TypeRegA {
+			slugSize += hdr.Size
+		}
+	}
+
+	// baz.txt would normally be ignored, but should not be
+	var bazFound bool
+	for _, file := range fileList {
+		if file == "baz.txt" {
+			bazFound = true
+		}
+	}
+	if !bazFound {
+		t.Fatal("expected file baz.txt to be present, but not found")
+	}
+
+	// .terraform/file.txt would normally be ignored, but should not be
+	var dotTerraformFileFound bool
+	for _, file := range fileList {
+		if file == ".terraform/file.txt" {
```

Comment:
It might be even more clear to write a `.terraformignore` file which explicitly ignores `baz.txt` or something. This works fine, but it relies on the current defaults.

Reply:
This test slugs up `testdata/archive-dir`, which does have an explicit `.terraformignore` that ignores `baz.txt`. The test that that file is not skipped is above this one, so this is testing the default rules as well as a rule created by `testdata/archive-dir/.terraformignore`.

We could write the `.terraformignore` file inline in this test, but that doesn't really seem better than using the `testdata` directory to me.

Reply:
Ah, I missed that the .terraformignore was already in the slug. I think that's good. Thanks for clarifying!